### PR TITLE
fix(types): Update model.d.ts to give more correct return type for `.toJSON()`

### PR DESCRIPTION
**Pull request by JonathanWolfe**
### Description of change

Change the return type of a model's `.toJSON()` method from generic `object` to the more specifc and helpful `TModelAttributes`.

### CODE REVIEW

**Comment by JonathanWolfe** [comment]
CI seems to have failed from a deadlock, but that has to be entirely unrelated to my changes because I only edited the type defs 😂

**Comment by papb** [comment]
Hello! I see you are a first-time contributor, thank you for taking the time to help Sequelize! I hope to see more PRs from you in the future!

Please add at least one test to make sure your implementation works as intended and to prevent regressions in the future. For good examples on how to test typescript typings, check the following PRs: [11368](https://github.com/sequelize/sequelize/pull/11368/files), [11379](https://github.com/sequelize/sequelize/pull/11379/files), [11520](https://github.com/sequelize/sequelize/pull/11520/files), [13042](https://github.com/sequelize/sequelize/pull/13042/files). You can also use [expect-type](https://www.npmjs.com/package/expect-type#documentation) if you need more complex assertions for your test.

**Comment by papb** [comment]
> 
> 
> CI seems to have failed from a deadlock, but that has to be entirely unrelated to my changes because I only edited the type defs 😂

Oh that deadlock was happening randomly with MySQL but I've fixed that now, won't bother again haha

**Comment by JonathanWolfe** [comment]
@papb I added the tests like you requested. Give it another look over

File: types/lib/model.d.ts [hunk 1]
```diff
@@ -2820,7 +2820,7 @@ export abstract class Model<TModelAttributes extends {} = any, TCreationAttribut
    * Convert the instance to a JSON representation. Proxies to calling `get` with no keys. This means get all
    * values gotten from the DB, and apply all custom getters.
    */
-  public toJSON(): object;
+  public toJSON(): TModelAttributes;
```
Comment (Nkmol) [inline types/lib/model.d.ts:2823]: Might be outside of the scope of this PR, but we could possibly even extend this type to use any generic specialization.

```
public toJSON<T extends TModelAttributes>(): T
```

This makes it possible to have a bit more control over the output when you want a very specific (derived) type. For example the case where certain properties are non-null or `Required`.

**Comment by liSong5713** [comment]
waiting for this pr. it`s important for know  definitions

**Pull request closed without merging**